R/target_market_share.R

```diff
@@ -0,0 +1,195 @@
+#' Add targets for production, using the market share approach
+#'
+#' This function calculates the portfolio-level production targets, as
+#' calculated using either the technology market share ratio `tmsr` or sector
+#' market share percentage `smsp`.
+#'
+#' @param data A "data.frame" like the output of
+#'   [r2dii.match::prioritize()].
+#' @param ald An asset level dataframe like [r2dii.data::ald_demo].
+#' @param scenario A scenario dataframe like [r2dii.data::scenario_demo_2020].
+#' @param region_isos A dataframe like [r2dii.data::region_isos] (default).
+#' @param use_credit_limit Logical vector of length 1. `FALSE` defaults to using
+#'   the column `loan_size_outstanding`. Set to `TRUE` to use the column
+#'   `loan_size_credit_limit` instead.
+#' @param by_company Logical vector of length 1. `FALSE` defaults to outputting
+#' `weighted_production_value` at the portfolio-level. Set to `TRUE` to output
+#' `weighted_production_value` at the company-level.
+#'
+#' @return A tibble with the same groups as the input (if any) and columns:
```

Comment:
The output is different when the value of `by_company` is `FALSE` versus `TRUE`.
It this is not a bug but a feature, then we need to document it; maybe in `@return`, in
`@param`, and maybe comment this in `@examples`.

``` r
library(dplyr, warn.conflicts = FALSE)
library(r2dii.banks)
#> Loading required package: r2dii.analysis
#> Loading required package: r2dii.data
#> Loading required package: r2dii.match

data <- r2dii.data::loanbook_demo %>%
  r2dii.match::match_name(r2dii.data::ald_demo) %>%
  r2dii.match::prioritize()

ald <- r2dii.data::ald_demo
scenario <- r2dii.data::scenario_demo_2020
region_isos <- r2dii.data::region_isos_demo

out1 <- data %>% 
  target_market_share(
    ald = ald, scenario = scenario, region_isos = region_isos,
    # default
    by_company = FALSE
  )

out2 <- data %>% 
  target_market_share(
    ald = ald, scenario = scenario, region_isos = region_isos,
    by_company = TRUE
  )

match(names(out2), names(out1))
#> [1]  1  2  3  4 NA  5  6

setdiff(names(out2), names(out1))
#> [1] "name_ald"
```

<sup>Created on 2020-06-12 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Reply:
ya this is a feature. not sure the best way to document this?

Reply:
I attempted to concisely describe how `by_company = TRUE` will output the additional column `name_ald` (we are effectively adding `name_ald` as a grouping variable). Let me know if there is a better way to explain this!